### ISSUE TITLE
How to Migrate from Consumption Plan to App Service Plan (and vice versa)

### ISSUE THREAD

**Issue by lindydonna** [Azure/Azure-Functions#155]
_From @stephlocke on November 30, 2016 12:4_

Is the migration path:
1. Make sure you have all your keys
2. Delete existing Function App
3. Delete Service Plan
3. Wait for DNS to flush
4. Create new Function App on Consumption Plan with same name and same storage account
5. Configure CI, app settings etc

Will this work with all the history so blob's don't get reprocessed etc?

_Copied from original issue: Azure/azure-webjobs-sdk-script#1010_

**Comment by lindydonna** [comment]
@stephlocke This is better asked on the MSDN forums or StackOverflow (you'll have more folks seeing and answering questions there, as opposed to GitHub isuses).

Are you using blob or queue triggers?  Those behave differently in terms of how multiple hosts will process.

@christopheranderson Do you know the answer to this question?

**Comment by lindydonna** [comment]
_From @stephlocke on December 2, 2016 20:51_

IMO, this should be a docs thing as people will need to do this for maybe
the next year?

**Comment by lindydonna** [comment]
_From @stephlocke on December 5, 2016 11:44_

I'm using blob triggers, and also eventhub triggers elsewhere that would need migrating also

**Comment by lindydonna** [comment]
This will become a real feature soon, so don't need to document.

**Comment by jorgechen** [comment]
@lindydonna Any updates on this feature to automatically migrate from Consumption to App Service Plan?

**Comment by aherrick** [comment]
Would also like an update on this! Is there at least a manual process?

**Comment by jmarbutt** [comment]
I need an update on this also

**Comment by JensMadsen** [comment]
me too

**Comment by leftler** [comment]
@lindydonna I would also like a update, it has been several months since we heard anything about this.

**Comment by valdisiljuconoks** [comment]
she is not working in MS anymore..

**Comment by stephlocke** [comment]
@christopheranderson are you able to pick this up? Still a big gaps in the docs / functionality that is still giving people trouble

**Comment by lindydonna** [comment]
CC @jeffhollan

**Comment by syedhassaanahmed** [comment]
+1

**Comment by neilhosey** [comment]
+1

**Comment by mjirhage** [comment]
+1

**Comment by mharr** [comment]
+1

**Comment by talkwaqar** [comment]
+1

**Comment by patmci** [comment]
+1

**Comment by leftler** [comment]
@jeffhollan can we get any kind of update on this, it has been two months without any kind of word from microsoft about this. Is it at least on the backlog?

**Comment by tence** [comment]
+1

**Comment by darrengillis** [comment]
+1

**Comment by ckarras** [comment]
+1

**Comment by ronWoodley** [comment]
+1

**Comment by Sohojoe** [comment]
+1

**Comment by harrymillerr** [comment]
+1

**Comment by mohanprasathsj** [comment]
+1

**Comment by Jhultink** [comment]
+1

**Comment by ahbrett** [comment]
+1

**Comment by jeffhollan** [comment]
Circling back on this.  Definitely an experience we still have on the backlog *however* it's not on the immediate backlog (read: slated for pre-July).  We have to do some work to make transition between consumption plan and a dedicated plan more automatic - still pending investigation on our side here.  These upvotes and interest are helpful in pushing that along though. It's not for lack of wanting on our part.

In the meantime we can document the different ways to move - @lindydonna had a good suggestion up top to leverage same name.  Could also re-create a new function app and migrate code to it. Depends largely on if HTTP triggered or not.  Not sure if still an appetite for documentation (nothing outside of what listed here and on top comment), or if most interest in this issue is on the "move" feature.  Let me know so we can track appropriately.

**Comment by jeffhollan** [comment]
Just one more note - we're discussing feasible of asks now (both in terms of documenting and making easier).  Would be good to know what folks expect or what would motivate someone to move from consumption to dedicated (or visa versa) and what hoping to achieve to make sure we solve for the right scenarios.  I have my own personal experiences/thoughts but always better to come straight from users :).  Any input / clarity on that would be great.

**Comment by darrengillis** [comment]
I had some HTTP triggered functions on a consumption plan but the delay in starting them up wasn't the best user experience for our particular situation.  Because of this, I wanted to move over to the app service plan.

**Comment by ahbrett** [comment]
Same as darrengillis above, the delay (up to 10 minutes) was too long for our needs.

**Comment by jorgechen** [comment]
@jeffhollan Thanks for owning this.  Hope to hear more updates since this is an important issue.
Regarding why there's a need to switch between Consumption Plan and App Service Plan, there are various reasons:

- During development, we might decide to test a AzF using the Consumption Plan then decide it is better running as always-on App Service Plan
- Businesses may want to switch plans to reduce op costs, so they switch from App Service to Consumption
- We've found that some Consumption Plan is unreliable due to cold starts

**Comment by harrymillerr** [comment]
I had a similar experience as Darren except it was on a ServiceBus Trigger, some messages didnt get processed until it had started up. Ended up moving them to WebJobs for the time being.

**Comment by mharr** [comment]
@jeffhollan Thanks for some response on this issue.

"Making easier" would definitely be better, and is needed.  Having to take your site down, rebuilding it, and reinstalling it just to change billing plans is quite a problem.  So just documenting that you have to destroy your site is certainly not a desirable solution.

@jorgechen has several good reasons for wanting/needing to switch.  I would like to add to his #2 item that an application with a seasonal variation for volume/traffic will want to change back and forth.  During seasonal heavy traffic periods, startup delays using consumption plan are manageable, because its happens less frequently.  But during slower times,  that spinup delay for every user becomes a support nightmare.  This might seem contra-indicative, as one would think being on consumption plan for lighter usage times would be more cost-effective, but that long warmup delays causes more problems than the savings for the always on mode.  

Bottom line, it will depend on application and business needs, and it would be very helpful if we could change without wiping out the site each time.

**Comment by jeffhollan** [comment]
This is all great. So I can likely correct statement after doing some initial discussions here that there may be opportunities we can improve this in this semesters roadmap. We’re meeting in a few days to discuss but this validates scenarios to help us target right fixes. Will update as I learn more in next 2-3 weeks for thoughts and timing.

**Comment by DavidJFowler** [comment]
Virtual Networks and Hybrid Connections are not available in the consumption plan.

**Comment by DeV1L** [comment]
It can be done easy:

1. Create new AppService plan
2. Open CloudShell (PoS)
3. Run command
`
Select-AzureRmSubscription -SubscriptionId "[id]" 
Set-AzureRmWebApp -Name "[function name]" -ResourceGroupName "[resource group]" -AppServicePlan "[new app service plan name]"
`

**Comment by MattFromGer** [comment]
I get the following error with your command:
`Set-AzureRmContext : A positional parameter cannot be found that accepts argument 'Set-AzureRmWebApp'`

**Comment by sirigere** [comment]
Thanks DeV1L, you command worked for me. I was able to move functions from consumption to regular app service plan.

**Comment by DeV1L** [comment]
@MattFromGer , sorry, commands merged to one line :) 
I've fixed it in the comment

**Comment by gzepeda** [comment]
As @DavidJFowler mentions, I need to migrate in order to consume resources on a VPN accessible through a VNET. VNETs are not supported while using Consumption Plans.

**Comment by m1nkeh** [comment]
ace, thanks for the PowerShell cmd! 🤓

**Comment by Architekt909** [comment]
If you're a noob to the azure command line stuff like me and are trying to follow @DeV1L instructions, just click the ">_" icon in your Azure Portal (top right corner, next to the gear/settings icon). Then change the shell from BASH (the default) to "PowerShell Preview" (it's the drop down that appears just above the console that opens up). The instructions worked perfectly, and I just switched from consumption to ASP, allowing me to set "always on" in my serverless functions. Thanks!

**Comment by jeffhollan** [comment]
And thanks @DeV1L for the script. It should be noted this powershell should work fine when moving from Consumption TO App Service Plan.  However when moving from App Service Plan to Consumption there may be an additional challenge in that consumption functions are backed by a storage account, and app service ones do not require one.  So an app-service-started function likely doesn't have an associated storage account and could have issues if moving out of an app service plan.

That said the most common migration does seem to be from Consumption into an App Service Plan in which case the PowerShell command (usable in the Azure Cloud Shell) seems like a great fit.

@ggailey777 as FYI as we should create an item to document this in the hosting plans doc for functions

**Comment by petrepopescu21** [comment]
@Architekt909 One would argue that running a Function App inside an App Service Plan will mean it's not serverless anymore.

**Comment by ggailey777** [comment]
@jeffhollan do we have the feature deltas between consumption and App Service plans documented anywhere? Seems like key info when deciding on plans.

**Comment by jeffhollan** [comment]
Just circling back on this - we have this feature currently planned as a simple operation for the next few months.  Plan to make it seamless to move from consumption to app service plan and back again.

**Comment by stephlocke** [comment]
I look forward to no longer seeing this in my github notifications! 🤣 Thanks @jeffhollan

**Comment by chrisparkeronline** [comment]
I now need this also. I guess a solution would be to remove the function and recreate it. Moving functionality would be much easier.

**Comment by Matthewsre** [comment]
I am also looking for this feature. 

Supporting Hybrid Connections from consumption plans would be ideal for my scenario and I found a feature request to vote on here for those interested:
https://feedback.azure.com/forums/355860-azure-functions/suggestions/33398398-add-support-for-hybrid-connections-to-consumption

**Comment by oughton** [comment]
Also holding out for this feature instead of manually migrating. Thanks Functions team!

**Comment by KirkMunro** [comment]
There's also the need to easily switch from one app service plan to another, so that you can measure performance and weigh the pros and cons of a higher cost ASP vs better performance gains that come from it when under load. If this is truly as easy as a PowerShell command, why not just make this available in the Azure UI?

**Comment by jeffhollan** [comment]
It should be noted there are a few things at play here.  There are differences in how Azure Functions on consumption and Azure Functions on app service plan work behind the scenes.  Secrets are stored differently, files are stored differently, and scale is handled differently.  There is work underway in the next release that will make it possible from an API standpoint to move between consumption and app service plan.  There are also changes in defaults so secrets and files should work if an app started as "consumption" and moved to "app service plan."  I don't know for sure how many combinations will 'just work', but from my understanding moving from consumption to app service plan should 'just work', and moving from app service plan to consumption will work if it initially started as consumption.  There may be some extra steps if the app was 'born' as an app service plan hosted and then is being moved to consumption.  

This is an item we want to improve (including portal UX), but has some steps involved and I can't make promises it's all going to work flawlessly even this calendar year.  Once I get a feel for what's possible after this next release I can update with more docs and examples, and hopefully we will have some Portal UX by end of the year (on the backlog for sure)

**Comment by jorupp** [comment]
I think we can all understand that sometimes something that looks complex on the surface is a bit harder under the covers :)  More blogs on "how this stuff actually works under the covers" is great too :)

What would be helpful to us is just having a clear picture of what we can do now (which may actually be in the docs now - haven't looked at the details recently), what is planned in the near term (w/ rough dates where possible - ie. the visibility we've been getting re: functions v2), and where it's heading long-term (again, rough dates (ie. quarters) are great, but we understand you can't always predict those things).

Sometimes it's hard for us to realize that something doesn't work - there's a lot of documentation out there, and sometimes we think that we must just be missing the thing saying how to do something - calling out the things that don't work (yet) can be just as helpful to guiding us towards the correct decisions for our projects as the docs explaining how to do things.

**Comment by KirkMunro** [comment]
Also regarding the PowerShell that was shared by @DeV1L earlier (thanks for sharing that), there are a few important details that are worth noting:

1. When invoking Set-AzureRmWebApp, the -AppServicePlan parameter can be the app service plan name if everything is in the same region/resource group.
1. If the app service plan is in a different resource group than the Azure function, you can only switch to it if you pass in its id (which can be found by looking up the app service plan with Get-AzureRmAppServicePlan, and then passing in the Id property of the object you get back).
1. If the app service plan is in a different region than the Azure function, then you cannot switch to it using Set-AzureRmWebApp -- it returns an error indicating "Cannot change the site _________ to the App Service Plan _________ due to hosting constraints."

**Comment by BrendanGrant** [comment]
I can't yet identify the specific case @KirkMunro, however I've encountered a case where a consumption plan function, (now) in the same resource group as the target App Service Plan cannot be moved via Set-AzureRmWebApp... and all are in the same region with the "Cannot change the site _________ to the App Service Plan _________ due to hosting constraints." error.

**Comment by myusrn** [comment]
@jeffhollan where are things now on the function app portal UI support for migrating from consumption app service plan to a pay-as-you-go [payg] app service plan and preferably the ability to go back the other way also.  

Initially i'd be fine with azure-cli functionapp and powershell Set-AzureRmWebApp cmdlet, that @DeV1L shared earlier, command line options for going to payg and back from payg to consumption plan.

I have a project where i'm having to do web perf and load test against default of function app, api app nd vm app deployment options and then compare results against incremental scale out of each of those.  In function app case this means moving from the consumption plan up to first payg plan that provides scale out instances.  I expect i'll have to repeat these comparison tests which is why the ask to be able to go from function app that has been migrated to payg app service plan back to consumption app service plan.  I might expect other customers to have non-benchmarking scenarios where they moved from consumption to payg and then realized that didn't make a significant difference, or function app usage model changes, such that going back to consumption makes sense for them.

**Comment by nielsbrinch2** [comment]
As Azure Functions work differently on Consumption Plan and a an App Service Plan, I don't think investing in a migration makes sense. Some times it doesn't work and then what? Will support have to say that's by design, because they are different. No one will understand that.

As long as they are different, I would fully understand the migration doesn't work and won't get prioritized. Instead these two targeted features would help the migration and they would both work independently:

1. Transfer application settings from this App Service to that
2. Swap resource name (domain) between this App Service and that

Unfortunately the latter would depend on a "rename resource" which I do not think is possible. So how would a migration ever work anyway, one way or the other?

But would be a great feature! And it could also be used as a poor mans deployment slots for Functions.

**Comment by DavidJFowler** [comment]
> I can't yet identify the specific case @KirkMunro, however I've encountered a case where a consumption plan function, (now) in the same resource group as the target App Service Plan cannot be moved via Set-AzureRmWebApp... and all are in the same region with the "Cannot change the site _________ to the App Service Plan _________ due to hosting constraints." error.

This can occur if the webspaces don't match:

https://social.msdn.microsoft.com/Forums/office/en-US/c498dd15-fb67-4ab3-8e3f-46b9a594dc87/unable-to-transfer-app-services-between-existing-service-plan-same-region-and-same-resource-group?forum=windowsazurewebsitespreview

**Comment by jeffhollan** [comment]
Ok let me at least give an update as this has gone a little stale and I've learned a lot / we've done some work:

1. You can change the plan type by modifying the "serverFarmId" property of the Azure Functions Microsoft.Web/sites resource (via something like the `az resource` CLI command or [https://resources.azure.com](https://resources.azure.com)).  
1. Note that *by default* apps created in an App Service Plan do not have two important properties: WEBSITE_CONTENTAZUREFILECONNECTIONSTRING and WEBSITE_CONTENTSHARE that consumption functions need to scale.  These don't hurt to have in an app service plan but you *must* have them in a consumption plan

So, to move from consumption -> dedicated for now you can use a manual method.  From dedicated -> consumption it **may** work but make sure you have right app settings first.

We do have some UX and some features coming out in the next month that will allow migration between plan types in the portal

**Comment by gpazo** [comment]
@jeffhollan Any updates on adding this to the UI? I tried the commands above by @DeV1L but no luck:

`Set-AzureRmWebApp : Operation returned an invalid status code 'NotFound'`

I'm attempting to run the commands in the same region migrating from consumption -> dedicated

**Comment by adetayorabiu** [comment]
> It can be done easy:
> 
> Create new AppService plan
> Open CloudShell (PoS)
> Run commands
> Select-AzureRmSubscription -SubscriptionId "[id]" 
> Set-AzureRmWebApp -Name "[function name]" -ResourceGroupName "[resource group]" -AppServicePlan "[new app service plan name]"

Hi @DeV1L  I get this error when I run this. 

Set-AzureRmWebApp : Operation returned an invalid status code 'Conflict'
At line:1 char:1
+ Set-AzureRmWebApp -Name my-fnc-name -ResourceGroupName my-rg-name
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : CloseError: (:) [Set-AzWebApp], DefaultErrorResponseException
+ FullyQualifiedErrorId : Microsoft.Azure.Commands.WebApps.Cmdlets.WebApps.SetAzureWebAppCmdlet

@jeffhollan The option to modify **serverFarmId** via https://resources.azure.com wasn't successful either.   

Please help!

**Comment by gpazo** [comment]
@adetayorabiu +1 modifying the **serverFarmId** using resource explorer caused the page to refresh with the **serverFarmId** property unchanged

**Comment by jeffhollan** [comment]
The only officially supported way today to move from consumption to a dedicated app service plan would be to re-deploy.  The differences in file and app data storage are technically navigable, but if anyone is stuck I'd recommend redeploying.

Note that there are some limitations if trying to change your app service plan regardless if functions.  Things like trying to change the [region](https://docs.microsoft.com/en-us/azure/app-service/app-service-plan-manage#move-an-app-to-a-different-region) aren't supported.  It's also important to note the note [here on webspaces](https://docs.microsoft.com/en-us/azure/app-service/app-service-plan-manage#move-an-app-to-another-app-service-plan), where even two plans in the same region may not be compatible to move without re-deploy.  The way I tested was with two plans in the same resource group so I knew they were in the same webspace.

We are working on easier ways to move from a consumption to more dedicated plan shortly, and do have some UX planned to support that, but it will be limited in scope as well.

**Comment by pdemro** [comment]
> We do have some UX and some features coming out in the next month that will allow migration between plan types in the portal

@jeffhollan Have these updates been deployed?  I am attempting to "scale up" a function app this morning and do not see the option to change or modify the subscription

**Comment by leonheess** [comment]
When will this feature arrive?

**Comment by jeffhollan** [comment]
See this related issue: https://github.com/Azure/Azure-Functions/issues/1194#issuecomment-580416039

**Comment by stephlocke** [comment]
Still for me the main point of concern is avoiding a retriggering for already processed stuff. If we had a way of importing some history or state, for me things would be fine 👍

**Comment by IanKemp** [comment]
Come on Microsoft...

When will this basic functionality be added...

Will it be sometime this millennium?

**Comment by IanKemp** [comment]
When you create a support ticket in Azure about moving a function app, it actually links to this GitHub issue.

I created a ticket anyway because if Microsoft is happy to waste my time by not fixing this issue for **THREE WHOLE YEARS**, I'm happy to waste their time. And will keep submitting a new ticket every time I have this issue and it's not fixed.

Pull your finger out, Microsoft.

**Comment by jeffhollan** [comment]
Details on progress in this comment https://github.com/Azure/Azure-Functions/issues/1194#issuecomment-580416039

**Comment by stephlocke** [comment]
On that issue, it's unclear whether that supports state/history migration when you change plans. I had the situation where my function had already processed millions of blobs and I wouldn't want the change in plan to cause all those blobs to get re-processed.

**Comment by jeffhollan** [comment]
The state for functions is stored in `AzureWebJobsStorage` - so assuming you keep the same `AzureWebJobsStorage` azure storage account (which I would recommend), it should preserve things like checkpoints.

**Comment by stephlocke** [comment]
So my point 5 on the original issue was right? Just point at the same storage and easy peasy? 😱

**Comment by chetanmotamarri** [comment]
> Also regarding the PowerShell that was shared by @DeV1L earlier (thanks for sharing that), there are a few important details that are worth noting:
> 
> 1. When invoking Set-AzureRmWebApp, the -AppServicePlan parameter can only be the app service plan name if everything is in the same region/resource group.
> 2. If the app service plan is in a different resource group than the Azure function, you can only switch to it using Set-AzureRmWebApp if you pass in the app service plan id (which can be found by looking up the app service plan with Get-AzureRmAppServicePlan, and then passing in the Id property of the object you get back).
> 3. If the app service plan is in a different region than the Azure function, then you cannot switch to it using Set-AzureRmWebApp -- it returns an error indicating "Cannot change the site _________ to the App Service Plan _________ due to hosting constraints."

This solution is not working. I tried to change from Consumption plan to App service Premium plan. It worked for a day or two, after that my azure function got reverted back to original consumption plan.

**Comment by span** [comment]
> It can be done easy:
> 
>     1. Create new AppService plan
> 
>     2. Open CloudShell (PoS)
> 
>     3. Run commands
>        `Select-AzureRmSubscription -SubscriptionId "[id]" `
>        `Set-AzureRmWebApp -Name "[function name]" -ResourceGroupName "[resource group]" -AppServicePlan "[new app service plan name]" `

Here is how to do it with az cli:

`az resource update --resource-type "Microsoft.Web/sites/config" --name "[function name]" --resource-group "[resource group]" --set serverFarmId="[new app service plan name]"`

**Comment by jeffhollan** [comment]
Just updating this comment - the work so that virtually all consumption functions should be able to move to a premium plan in the same resource group has finished rolling out.

**Comment by justinimel** [comment]
I am unable to run these commands in PowerShell or Bash.

Powershell gives this error:
Set-AzWebApp: Operation returned an invalid status code 'BadRequest'

Bash gives this error:
Property 'serverFarmId' not found on root. Send it as an additional property .
The request content was invalid and could not be deserialized: 'Could not find member 'serverFarmId' on object of type 'ResourceDefinition'. Path 'serverFarmId'.'.

Have the commands changed? Everything is in the same resource groups and regions.

**Comment by nexxai** [comment]
I'm getting the same issue when trying to do:

```
➜  ~ az resource update --resource-type "Microsoft.Web/sites" --name "testing-move" --resource-group "testing-asp" --set serverFarmId="testing-destination"
Property 'serverFarmId' not found on root. Send it as an additional property .
The request content was invalid and could not be deserialized: 'Could not find member 'serverFarmId' on object of type 'ResourceDefinition'. Path 'serverFarmId'.'.
```

**Comment by nexxai** [comment]
I'm getting the same issue.

In Bash:

```
➜  ~ az resource update --resource-type "Microsoft.Web/sites" --name "testing-move" --resource-group "testing-asp" --set serverFarmId="testing-destination"
Property 'serverFarmId' not found on root. Send it as an additional property .
The request content was invalid and could not be deserialized: 'Could not find member 'serverFarmId' on object of type 'ResourceDefinition'. Path 'serverFarmId'.'.
```

In PowerShell:

```
PS /home/me> Set-AzureRmWebApp -Name "testing-move" -ResourceGroupName "testing-asp" -AppServicePlan "testing-destination"
Set-AzWebApp: Operation returned an invalid status code 'BadRequest'
```

@jeffhollan Any ideas?

**Comment by DanielHolth** [comment]
Bash did not work, but PowerShell worked. Provided all 3 components exists in the same region (West-Europe in my case)

**Comment by IanKemp** [comment]
> Bash did not work, but PowerShell worked. Provided all 3 components exists in the same region (West-Europe in my case)

Did you use `az.exe` or the `Set-AzWebApp` cmdlet?

**Comment by DanielHolth** [comment]
Set-AzureRmWebApp -Name "..." -ResourceGroupName "..." -AppServicePlan "..."

**Comment by rhythmnewt** [comment]
Neither of the commands worked for me with the same typical messages. Tried consumption plan to existing app service plan.
`The request content was invalid and could not be deserialized: 'Could not find member 'serverFarmId' on object of type 'ResourceDefinition'. Path 'serverFarmId'.'.`

and 
`Set-AzWebApp: Operation returned an invalid status code 'BadRequest'`

**Comment by tdwright** [comment]
Hi @jeffhollan. Hope you're well?

Pleased to hear that
>  the work so that virtually all consumption functions should be able to move to a premium plan in the same resource group has finished rolling out

Apologies if I've missed it (in this now rather long issue), but are there any plans afoot to support the other direction? I.e. to move from a functions app backed by an app service plan to a consumption based plan?

**Comment by dominikfoldi** [comment]
I was able to move back my Function from App Service Plan to Consumption plan by using the https://resources.azure.com/ website.

Enter edit mode, find your Function App and change the following:
```
{
  serverFarmId: "..../<your-consumption-app-service-name>"
  ...
  // Scroll a bit down
  sku: "Dynamic"
}
```

Update the resource and you are done.

**Make sure that you turn Always On off in the App Service Plan settings before the resource update to make this work.**

**Comment by msft-noorjaber** [comment]
Hey @dominikfoldi , that's not working now (at least for me), the PUT operation will succeed, however if you do a GET, the SKU will be the old one.

**Comment by Gamecock** [comment]
This is still broken from the command line:
PowerShell    Set-AzureRmWebApp    returns a 'NotFound" for my function app.  I retyped many times, tied both quotes and unquoted.
Bash:  az resource update returns:  Property 'serverFarmId' not found on root. Send it as an additional property .
@dominikfoldi  answer for ResourceExplorer worked,I did not update "sku:" and it updated itself to PremiumV2.

**Comment by jobullen** [comment]
The az CLI is not correct. Use the following command to do the move. 

```
az resource update \
    --resource-type "Microsoft.Web/sites" \
    --name "[function app name]" \
    --resource-group "[resource group]" \
    --set properties.serverFarmId="[new app service plan name]"
```

You can confirm the property in the REST API and it is under properties.serverFarmId.

[Web Apps - Create Or Update](https://docs.microsoft.com/en-us/rest/api/appservice/web-apps/create-or-update#request-body)

**Comment by fzoccara** [comment]
I hope we solved re-creating all azure function with windows instead of linux, not happy about this, but it seems working now.

take care
fra

**Comment by prenaissance** [comment]
Did anyone have their consumption based function app automatically changed to B1 plan? :(

**Comment by dvalenzuela-com** [comment]
@prenaissance probably too late for you but I managed to do it without any issues, what I definitely did not expect. 

What I did:

* Go to this site https://learn.microsoft.com/en-us/powershell/module/az.websites/set-azwebapp?view=azps-13.0.0#code-try-2 
* Use the azure shell console to the right
* Selected PowerShell
* Used the command given above by somebody, `Set-AzureRmWebApp -Name func-XXX-dev-westeu-101  -ResourceGroupName rg-XXX-dev-westeu-001 -AppServicePlan asp-XXX-dev-westeu-101`
* waited a bit and that was it.
